### DISCUSSION TITLE
Use Unleash for Messaging

### DISCUSSION THREAD

**Discussion by omarzouk** [Unleash/unleash#2108] [Q&A]
Hi, we think unleash fits a lot of our needs, however we were wondering if we could also use it for Messaging.

We think that the [toggle variants payload](https://docs.getunleash.io/advanced/toggle_variants) could be a feasible place to hold messaging data, that is then shown to users who meet the constraints.

However, the only issue is we would want to acknowledge a message as "seen" on the backend, so if a user re-installs the app, they wouldn't get the messages triggered again.

What would be a good way to push back "seen" data? we thought of adding the userId to a Segment using a [UserIDs](https://docs.getunleash.io/user_guide/activation_strategy#userids) constraint, however the wording on the wiki makes it sound like this is meant to be a small list.

**Comment by pransh15** [comment]
This is a very interesting usecase, @omarzouk! 🤔 
I do not have the answer to this question honestly, but I can invite discuss with the team and see what we can do here. Someone will get to you shortly! 😄

**Comment by ivarconr** [comment]
> wording on the wiki makes it sound like this is meant to be a small list.

yes this is kind of true. The reasoning being how we implemented the feature. We rewrite all segments to strategy constraints when the SDK does not understand segments. This have some limitations when the list of values get long (it kind of explodes by being expanded in all places it is being used), and we have thus been carefully not promoting too long lists until all SDKs supports segments (most do now) and we have confidence that sufficient amount of users has upgraded their SDKs (not there yet). 

Regardless, I would be careful if you plan to have more than thousand of IDs in there, then you are probably not using it for the right thing. Maybe you are are looking for a database instead?
